src/_pytest/tmpdir.py

```diff
@@ -26,7 +26,9 @@ class TempPathFactory:
         # using os.path.abspath() to get absolute path instead of resolve() as it
         # does not work the same in all platforms (see #4427)
         # Path.absolute() exists, but it is not public (see https://bugs.python.org/issue25012)
-        converter=attr.converters.optional(lambda p: Path(os.path.abspath(str(p))))
+        converter=attr.converters.optional(
+            lambda p: Path(os.path.abspath(str(p)))  # type: ignore
```

Comment:
what's the type error here?

Reply:
I'll add more extensive code comments explaining each `type: ignore`.